archive/imo/imo1962_q4.lean

```diff
@@ -0,0 +1,139 @@
+/-
+Copyright (c) 2020 Kevin Lacker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Lacker
+-/
+import data.real.pi
+
+open real
+noncomputable theory
+
+notation `π` := real.pi
+
+/-!
+Solve the equation `cos x ^ 2 + cos (2 * x) ^ 2 + cos (3 * x) ^ 2 = 1`.
+
+Since Lean does not have a concept of "simplest form", we just express what is
+in fact the simplest form of the set of solutions, and then prove it equals the set of solutions.
+-/
+
+def problem_equation (x : ℝ) : Prop := cos x ^ 2 + cos (2 * x) ^ 2 + cos (3 * x) ^ 2 = 1
+
+def solution_set : set ℝ :=
+{ x : ℝ | ∃ k : ℤ, x = (2 * ↑k + 1) * π / 4 ∨ x = (2 * ↑k + 1) * π / 6 }
+
+/-
+The key to solving this problem simply is that we can rewrite the equation as
+a product of terms, shown in `alt_formula`, being equal to zero.
+-/
+
+def alt_formula (x : ℝ) : ℝ := cos x * (cos x ^ 2 - 1/2) * cos (3 * x)
+
+lemma cosx2 {x : ℝ} : cos x ^ 2 = 1 - sin x ^ 2 :=
```

Comment:
Even though this is just a trivial rearrangement of a lemma already in mathlib, the fact that the corresponding `sin` lemma is in mathlib proper is enough reason for this one to be in mathlib proper as well. Note: it should be there for both real and complex arguments (you can probably deduce the real version of such a lemma from the complex one with a single call to `exact_mod_cast`, and if you can't then maybe some `norm_cast` lemmas are missing).

Reply:
OK